init_pose:
    joint_states:
        joint1: 0.000000
        joint2: 0.012272
        joint3: 0.036816
        joint4: 0.010738
    position :
        x: 0.287194
        y: 0.000000
        z: 0.190578
    quaternion :
        x: 0.000000
        y: 0.029908
        z: 0.000000
        w: 0.999553
    euler :
        roll: 0.000000
        pitch: 0.059825
        yaw: 0.000000
    # Matrix Rotation :
    #     [0.998211, 0, 0.0597896]
    #     [0, 1, 0]
    #     [-0.0597896, 0, 0.998211]

home_pose:
    joint_states:
        joint1: 0.000000
        joint2: -1.043107
        joint3: 0.358952
        joint4: 0.702563
    position :
        x: 0.135569
        y: 0.000000
        z: 0.237739
    quaternion :
        x: 0.000000
        y: 0.009204
        z: 0.000000
        w: 0.999958
    euler :
        roll: 0.000000
        pitch: 0.018408
        yaw: 0.000000
    # Matrix Rotation :
    #     [0.999831, 0, 0.0184068]
    #     [0, 1, 0]
    #     [-0.0184068, 0, 0.999831]

save_pose:
    joint_states:
        joint1: 0.001534
        joint2: -1.952758
        joint3: 1.257864
        joint4: 0.702563
    position :
        x: 0.105522
        y: 0.000143
        z: 0.129491
    quaternion :
        x: -0.000003
        y: 0.003835
        z: 0.000767
        w: 0.999992
    euler :
        roll: -0.000000
        pitch: 0.007670
        yaw: 0.001534
    # Matrix Rotation :
    #     [0.999969, -0.00153398, 0.00766978]
    #     [0.00153394, 0.999999, 1.17653e-05]
    #     [-0.00766979, -4.40457e-20, 0.999971]

find_object_pose:
    joint_states:
        joint1: 1.497165
        joint2: -0.948000
        joint3: 0.039884
        joint4: 1.784020
    position :
        x: 0.016929
        y: 0.066817
        z: 0.171629
    quaternion :
        x: -0.288633
        y: 0.310708
        z: 0.616367
        w: 0.663508
    euler :
        roll: 0.000000
        pitch: 0.875903
        yaw: 1.497165
    # Matrix Rotation :
    #     [0.0471036, -0.99729, 0.0565065]
    #     [0.638569, 0.0735645, 0.766041]
    #     [-0.768122, 0, 0.640303]

find_object_pose_new:
    joint_states:
        joint1: 1.569262
        joint2: -0.645806
        joint3: -0.487806
        joint4: 1.935884
    position :
        x: 0.012126
        y: 0.082211
        z: 0.214918
    quaternion :
        x: -0.275888
        y: 0.276312
        z: 0.650476
        w: 0.651474
    euler :
        roll: 0.000000
        pitch: 0.802272
        yaw: 1.569262
    # Matrix Rotation :
    #     [0.0010662, -0.999999, 0.00110281]
    #     [0.695074, 0.00153394, 0.718936]
    #     [-0.718937, 0, 0.695075]


before_grasp_pose:
    joint_states:
        joint1: 1.570796
        joint2: -0.541495
        joint3: -0.150330
        joint4: 1.492563
    position:
        x: 0.012000
        y: 0.137802
        z: 0.187212
    quaternion:
        x: -0.275601
        y: 0.275601
        z: 0.651187
        w: 0.651187
    euler:
        roll: 0.0
        pitch: 0.800738
        yaw: 1.570796
    # Matrix Rotation :
    #     [-3.04309e-08, -1, -3.13791e-08]
    #     [0.696177, -4.37114e-08, 0.71787]
    #     [-0.71787, -1.11022e-16, 0.696177]

init_left_pose:
    joint_states:
        joint1: 1.547787
        joint2: 0.013806
        joint3: 0.038350
        joint4: 0.013806
    position :
        x: 0.018334
        y: 0.275249
        z: 0.189387
    quaternion :
        x: -0.023047
        y: 0.023583
        z: 0.698545
        w: 0.714806
    euler :
        roll: -0.000000
        pitch: 0.065961
        yaw: 1.547787
    # Matrix Rotation :
    #     [0.0229576, -0.999735, 0.00151651]
    #     [0.997561, 0.0230076, 0.0658959]
    #     [-0.0659134, -6.93889e-18, 0.997825]

home_left_pose:
    joint_states:
        joint1: 1.550855
        joint2: -1.043107
        joint3: 0.388097
        joint4: 0.711767
    position :
        x: 0.014505
        y: 0.125606
        z: 0.230077
    quaternion :
        x: -0.019863
        y: 0.020263
        z: 0.699739
        w: 0.713834
    euler :
        roll: -0.000000
        pitch: 0.056757
        yaw: 1.550855
    # Matrix Rotation :
    #     [0.0199083, -0.999801, 0.00113116]
    #     [0.998191, 0.0199404, 0.0567156]
    #     [-0.0567268, -6.93889e-18, 0.99839]

place_object_pose:
    joint_states:
        joint1: 1.549321
        joint2: 0.412641
        joint3: 0.655010
        joint4: -1.041573
    position :
        x: 0.017563
        y: 0.259006
        z: 0.072213
    quaternion :
        x: -0.009120
        y: 0.009318
        z: 0.699414
        w: 0.714598
    euler :
        roll: -0.000000
        pitch: 0.026078
        yaw: 1.549321
    # Matrix Rotation :
    #     [0.0214667, -0.999769, 0.000559929]
    #     [0.999429, 0.021474, 0.0260687]
    #     [-0.0260748, -2.60209e-17, 0.99966]

after_place_pose:
    joint_states:
        joint1: 1.549321
        joint2: 0.042951
        joint3: 0.179476
        joint4: -0.641204
    position :
        x: 0.017702
        y: 0.265470
        z: 0.227234
    quaternion :
        x: 0.145394
        y: -0.148550
        z: 0.684196
        w: 0.699049
    euler :
        roll: -0.000000
        pitch: -0.418777
        yaw: 1.549321
    # Matrix Rotation :
    #     [0.0196183, -0.999769, -0.00873225]
    #     [0.913376, 0.021474, -0.406549]
    #     [0.406643, -5.55112e-17, 0.913587]


